consensus/src/consensus/remote_event_dispatcher.rs

```diff
@@ -253,25 +274,38 @@ impl<N: Network> Future for RemoteEventDispatcher<N> {
                     for txn in transactions {
                         let txn = txn.get_raw_transaction();
 
-                        // Process transaction senders
-                        if let Some(peers) = self.state.read().get_peers(&txn.sender) {
-                            for peer in peers {
-                                if let Some(receipts) = peer_receipts.get_mut(&peer) {
-                                    receipts.push((txn.hash(), block.block_number()))
-                                } else {
-                                    peer_receipts
-                                        .insert(peer, vec![(txn.hash(), block.block_number())]);
-                                }
-                            }
-                        }
+                        // Process transaction sender
+                        self.add_notification_receipts(
+                            &txn.sender,
+                            txn.hash(),
+                            block.block_number(),
+                            &mut peer_receipts,
+                        );
+
                         // Process transaction recipients
-                        if let Some(peers) = self.state.read().get_peers(&txn.recipient) {
-                            for peer in peers {
-                                if let Some(receipts) = peer_receipts.get_mut(&peer) {
-                                    receipts.push((txn.hash(), block.block_number()))
-                                } else {
-                                    peer_receipts
-                                        .insert(peer, vec![(txn.hash(), block.block_number())]);
+                        self.add_notification_receipts(
+                            &txn.recipient,
+                            txn.hash(),
+                            block.block_number(),
+                            &mut peer_receipts,
+                        );
+
+                        // Process staking transaction (which are a special case)
+                        if txn.recipient_type == AccountType::Staking {
+                            // Parse transaction data.
```

Comment:
```suggestion
                            // Parse transaction data
```
For consistency with the rest of the comments?